2/contrib/jenkins/kube-slave-common.sh

```diff
@@ -72,10 +72,8 @@ fi
 # generate_kubernetes_config generates a configuration for the kubernetes plugin
 function generate_kubernetes_config() {
     [ -z "$oc_cmd" ] && return
-    [ ! has_service_account ] && return
-    local crt_contents=$(cat "${KUBE_CA}")
-    # do not dump contents of cert here, but allow any error messages to be seen
-    openssl x509 -in $KUBE_CA > /dev/null
+    [ ! has_service_account ] && return   
+    local crt_contents=$(openssl x509 -in "${KUBE_CA}")
```

Comment:
not sure what this does...looks like it's going to reformat the cert before sticking it in the config.xml though?

Reply:
If you have a KUBE_CA file that looks like 

```
Bag Attributes
    localKeyID: 57 4A 32 C7 C2 B8 9E D9 E0 61 13 0C 9D 2A F9 6C A6 17 43 A0 
    friendlyName: My <Weird> Certificate
subject=/C=XX/L=Default City/O=Default Company Ltd
issuer=/C=XX/L=Default City/O=Default Company Ltd
-----BEGIN CERTIFICATE-----
MIIEBDCCAuygAwIBAgIBATANBgkqhkiG9w0BAQQFADBcMQswCQYDVQQGEwJVUzEL
MAkGA1UECAwCTkMxEDAOBgNVBAcMB1JhbGVpZ2gxEDAOBgNVBAoMB1JlZCBIYXQx
DDAKBgNVBAsMA0FQUzEOMAwGA1UEAwwFcmJvc3QwHhcNMTcxMTA5MTUzODE3WhcN
MjcxMTA3MTUzODE3WjBMMQswCQYDVQQGEwJVUzELMAkGA1UECAwCTkMxEDAOBgNV
BAoMB1JlZCBIYXQxDDAKBgNVBAsMA0FQUzEQMA4GA1UEAwwHQ2xpZW50MTCCASIw
DQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMQ4VctXYG8tozF77flbHe/YlX0q
Fo62lo4J/uRdnzjVFta/Lo6MedldnW/Wb2IrcODEOTPtqVAuvy9ag66HnWaO/DuW
C9HhyrsvzQR2oKHP7cIM7u4yI/xaeAWepBb/H8/oFBVxPTYViNn8zCW30BrUneh8
IYNY+mrl3uz5p2dq9oRW8HpE3RlArfMuCz5Cra+gQ5IdnsffaOqH8u54hluF4oOD
9nzHVSAnoMYs3g9/nt3DfgZ2uCbRQVqmEjDPuFyiRqRI2FK6b4t4JHzhBadiMSJB
/9ACFAaLaMe4xVWuODAZ2vEUuFDjmcoEL1cxw7PmGUxycIJ2lXJtjjVDK9ECAwEA
AaOB4DCB3TAJBgNVHRMEAjAAMB0GA1UdDgQWBBTJYU6/mDmFlQqUkgmmk120lB/J
ezCBjgYDVR0jBIGGMIGDgBQWLe5i+iMmbnI3vuI3n6X3tpbaC6FgpF4wXDELMAkG
A1UEBhMCVVMxCzAJBgNVBAgMAk5DMRAwDgYDVQQHDAdSYWxlaWdoMRAwDgYDVQQK
DAdSZWQgSGF0MQwwCgYDVQQLDANBUFMxDjAMBgNVBAMMBXJib3N0ggkA9Iru65rm
mkwwEwYDVR0lBAwwCgYIKwYBBQUHAwIwCwYDVR0PBAQDAgeAMA0GCSqGSIb3DQEB
BAUAA4IBAQAGPpH0U1OzQEGyZCogodj8Jak+QyrFYa7/Kt3Wmguxytj3La+Sgu7Q
RO3QCEm/wMfdjxhcoi8V+0nw/DwWa/ajC+pAgLLyw0OkzvSSgnzIDMzOUeMNulAZ
8U3V0ibaEOMt10xnI4dUr/jRiUg27f7SsmQm6BFlud3fSuTI/Wha8yZ51sGVY/LJ
R5t1WKw3g3aMXHUkKwajRR5X6/DS9dm7pKh+MOXMWJLS1iR01KZ/0YXCi6duQ+sY
sAcLrwqzg+PbMe+44DqFec7PPDRBtHR3j7FzIveJ5BRznemC5ZpU3nsBtB8cOGWw
9DSiXyMBKEin6iHBwq5RsCU/N8YeK6MQ
-----END CERTIFICATE-----
```

When you run the command, it will look like the *normal* form:
```
-----BEGIN CERTIFICATE-----
MIIEBDCCAuygAwIBAgIBATANBgkqhkiG9w0BAQQFADBcMQswCQYDVQQGEwJVUzEL
MAkGA1UECAwCTkMxEDAOBgNVBAcMB1JhbGVpZ2gxEDAOBgNVBAoMB1JlZCBIYXQx
DDAKBgNVBAsMA0FQUzEOMAwGA1UEAwwFcmJvc3QwHhcNMTcxMTA5MTUzODE3WhcN
MjcxMTA3MTUzODE3WjBMMQswCQYDVQQGEwJVUzELMAkGA1UECAwCTkMxEDAOBgNV
BAoMB1JlZCBIYXQxDDAKBgNVBAsMA0FQUzEQMA4GA1UEAwwHQ2xpZW50MTCCASIw
DQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMQ4VctXYG8tozF77flbHe/YlX0q
Fo62lo4J/uRdnzjVFta/Lo6MedldnW/Wb2IrcODEOTPtqVAuvy9ag66HnWaO/DuW
C9HhyrsvzQR2oKHP7cIM7u4yI/xaeAWepBb/H8/oFBVxPTYViNn8zCW30BrUneh8
IYNY+mrl3uz5p2dq9oRW8HpE3RlArfMuCz5Cra+gQ5IdnsffaOqH8u54hluF4oOD
9nzHVSAnoMYs3g9/nt3DfgZ2uCbRQVqmEjDPuFyiRqRI2FK6b4t4JHzhBadiMSJB
/9ACFAaLaMe4xVWuODAZ2vEUuFDjmcoEL1cxw7PmGUxycIJ2lXJtjjVDK9ECAwEA
AaOB4DCB3TAJBgNVHRMEAjAAMB0GA1UdDgQWBBTJYU6/mDmFlQqUkgmmk120lB/J
ezCBjgYDVR0jBIGGMIGDgBQWLe5i+iMmbnI3vuI3n6X3tpbaC6FgpF4wXDELMAkG
A1UEBhMCVVMxCzAJBgNVBAgMAk5DMRAwDgYDVQQHDAdSYWxlaWdoMRAwDgYDVQQK
DAdSZWQgSGF0MQwwCgYDVQQLDANBUFMxDjAMBgNVBAMMBXJib3N0ggkA9Iru65rm
mkwwEwYDVR0lBAwwCgYIKwYBBQUHAwIwCwYDVR0PBAQDAgeAMA0GCSqGSIb3DQEB
BAUAA4IBAQAGPpH0U1OzQEGyZCogodj8Jak+QyrFYa7/Kt3Wmguxytj3La+Sgu7Q
RO3QCEm/wMfdjxhcoi8V+0nw/DwWa/ajC+pAgLLyw0OkzvSSgnzIDMzOUeMNulAZ
8U3V0ibaEOMt10xnI4dUr/jRiUg27f7SsmQm6BFlud3fSuTI/Wha8yZ51sGVY/LJ
R5t1WKw3g3aMXHUkKwajRR5X6/DS9dm7pKh+MOXMWJLS1iR01KZ/0YXCi6duQ+sY
sAcLrwqzg+PbMe+44DqFec7PPDRBtHR3j7FzIveJ5BRznemC5ZpU3nsBtB8cOGWw
9DSiXyMBKEin6iHBwq5RsCU/N8YeK6MQ
-----END CERTIFICATE-----

```

and jenkins can consume it

Reply:
@bostrt made the observation in the bugzilla and I was testing it out (after answering your question regarding the empty field) when you reacted to my addressing your original concern

Reply:
Note, the openssl does not return 1 with the `Bag Attributes`, so I was not addressing the original issue

Other scenarios in my testing, like deleting one of the lines from --- begin cert -- to --- end cert -- did result in $? being 1

Reply:
@gabemontero Yea, that's more of what I was looking for. Openssl can server dual purposes: sanitizes certificate (stripping everything away but the cert itself) and validating the certificates contents.